backend/services/implementations/testService.ts

```diff
@@ -0,0 +1,32 @@
+import MgTest, { Test } from "../../models/test.model";
+import { CreateTestRequestDTO, SchoolTestDTO } from "../../types";
+import { getErrorMessage } from "../../utilities/errorUtils";
+import logger from "../../utilities/logger";
+import { ITestService } from "../interfaces/testService";
+
+const Logger = logger(__filename);
+
+class TestService implements ITestService {
+  /* eslint-disable class-methods-use-this */
+  async createTest(test: CreateTestRequestDTO): Promise<SchoolTestDTO> {
+    let newTest: Test;
+
+    try {
+      newTest = await MgTest.create({ ...test });
+    } catch (error) {
+      Logger.error(`Failed to create test. Reason = ${getErrorMessage(error)}`);
+      throw error;
+    }
+
+    return {
+      id: newTest.id,
+      name: newTest.name,
+      duration: newTest.duration,
+      admin: newTest.admin,
```

Comment:
i think you might need to validate that the `admin` id exists first. you can like call `getUserById()` to see if there's an error or not

Reply:
and then you can return the `adminDTO` instead of `admin` id like the other services -- but @joyce-shi maybe you can confirm?

Reply:
yup, i think reusing the existing function is a good idea!